gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java

```diff
@@ -83,16 +84,14 @@ public boolean hasNext() {
   @Override
   public DagTask next() {
     try {
-      DagActionStore.DagAction dagAction = this.dagActionQueue.take();  //`take` blocks till element is not available
-      // todo reconsider the use of MultiActiveLeaseArbiter
-      //MultiActiveLeaseArbiter.LeaseAttemptStatus leaseAttemptStatus = new MultiActiveLeaseArbiter.LeaseObtainedStatus(dagAction);
-      // todo - uncomment after flow trigger handler provides such an api
-      //Properties jobProps = getJobProperties(dagAction);
-      //flowTriggerHandler.getLeaseOnDagAction(jobProps, dagAction, System.currentTimeMillis());
-      //if (leaseAttemptStatus instanceof MultiActiveLeaseArbiter.LeaseObtainedStatus) {
-      // can it return null? is this iterator allowed to return null?
-      return createDagTask(dagAction, new MultiActiveLeaseArbiter.LeaseObtainedStatus(dagAction, System.currentTimeMillis()));
-      //}
+      MultiActiveLeaseArbiter.LeaseAttemptStatus leaseAttemptStatus = null;
+      DagActionStore.DagAction dagAction = null;
+//      while (!(leaseAttemptStatus instanceof MultiActiveLeaseArbiter.LeaseObtainedStatus)) {
+        dagAction = this.dagActionQueue.take();  //`take` blocks till element is not available
+//        multiActiveLeaseArbiter.tryAcquireLease(dagAction);
+//      }
+
+      return createDagTask(dagAction, (MultiActiveLeaseArbiter.LeaseObtainedStatus) leaseAttemptStatus);
```

Comment:
w/o the `while` loop above (that you commented out), we can't be sure this cast will succeed